### ISSUE TITLE
Officially: The actual benefit and eligible dates depend on the DAY of the birth month, which is not taken into account in the calculator

### ISSUE THREAD

**Issue by stkeros** [Gregable/social-security-tools#57]
A small point and one which is likely out of scope, but SS counts a birthday at the beginning of the month (e.g. Feb 1) different from others (Feb 2). These "edge" effects occur both with months, and with end of year issues. Here is a good explanation with examples:

https://secure.ssa.gov/poms.nsf/lnx/0300615015

**Comment by Gregable** [comment]
Ugh, that's an annoying corner case and I wasn't aware of it. I'm particularly annoyed because it requires collecting another bit of information from the user, which complicates the interface for the user as well.

I'll need to think about this a bit, but thanks for reporting.

**Comment by stkeros** [comment]
Another one which might be handled with an assumptions section. Collecting a birth DAY in addition to month and year increases complexity for the user. But also, if you try to explain what's going on with different dates of birth within the month, things get complicated. One of the best strengths for this tool is the "walkthrough" of the benefit/calculations. It's a great educational tool which is unique on the web. Adding certain details (birth day specific calcs) to the main input/output might decrease the readability and test the patience of all but the most savvy users, and lead to decreased overall educational value, on average.

**Comment by Gregable** [comment]
On the plus side, most people know by heart their own birth date and their spouses birthdate, so it's not asking for a lot in terms of data entry.

I do agree about the confusion. If someone inputs a birthday of Jan 15, 1950 and the calculator (correctly) shows them as 62 in Feb of 2012, that needs explanation or the user will assume a bug. That explanation itself requires more mental yoga for the user, which districts from the main point.

Still, I hate for the calculator to be incorrect, even if only in small ways. Grr.

**Comment by stkeros** [comment]
Yes it can indeed be confusing. Initially I thought there was a bug at open social security, but was just a quirk of the DOB thing. Pasted below is snippet from the output text, first with an April "1st" birthday, next with April 2nd second. As you can see the output is in "social security age" rather than lay person age. It confused me until I thought about it. So you are correct to worry about this as a (minor) issue. E.g. level of detail you want to deal with or explain.

Born on 1st of month:
![image](https://user-images.githubusercontent.com/25359648/41971603-070c5812-79dd-11e8-9c41-d65a1c2b4e98.png)

Born on 2nd of month:
![image](https://user-images.githubusercontent.com/25359648/41971621-185a2f22-79dd-11e8-990c-1eaa5ccd2942.png)

**Comment by Gregable** [comment]
I think I have this fully working now for both the primary earner and spousal birthdate selections. When you get a chance, please give this a spin and check that all of the dates make sense.

If the user selects a day-of-month that is greater than the 2nd, they will be presented with a "Special Rule" box explaining the English Law and the "entire month" rule. This isn't shown if the user has a birthdate on the 1st or 2nd of the month, since the calculations work as the user expects for this case, and explaining a rule that has no effect on them is not helpful.

**Comment by stkeros** [comment]
So it seems the 1985 demo is giving a 1st/2nd of the month b-day to self. I think the calculator works, but I still want to do some more math. Are you taking into account differences in benefits for those born on the 1st vs the 2nd?  See the chart pasted below, taken from: https://secure.ssa.gov/poms.nsf/lnx/0300615015

Note the year-end effects too. Which I will also try to go through and see if you've accounted for these.

It's a little weird that the slider ages don't match up for spouses of the same age. That is, the slider ages are in "social security" age. But if anyone reads the special rule they'll hopefully understand, and will only be evident for those with the same month/year date of birth but where one has a 1st/2nd b-day.

![image](https://user-images.githubusercontent.com/25359648/42342057-08585040-8063-11e8-83fc-c622684bf240.png)

**Comment by stkeros** [comment]
>  Are you taking into account differences in benefits for those born on the 1st vs the 2nd? 

Assuming the self demo has a 1st of month birthday, I see that the spouse of same age with a 2nd of month birthday has the same benefit if claiming early. I think they should have different early filing reductions.

**Comment by Gregable** [comment]
Yes, the demos all use 1st of month birthdays. My thought was no need to add confusion to the demos. I'll add the dates to the text to be more clear.

I guess I'm still confused. Looking at Bill and Cory, my understanding is that, due to english law:
 - Bill turns age X on 5/31/year
 - Cory turns age X on 6/1/year

However, both Bill and Cory have the same first month that they are age X, June. Cory is age X for the whole month of June because Cody was born on June 1. Bill isn't 'whole' in May, so we slide forward to June as well.

I don't follow why their reduction factor is the same.

**Comment by Gregable** [comment]
Ah, maybe it's because Full Retirement Age isn't affected by the "must be that age for the whole month" rule, only benefits?

Apologies for my dumb questions on this one.

**Comment by Gregable** [comment]
Let me see if I have this right:

In this scenario, Bill's 'english law' birthday is 5/31/35. Therefore bill's full retirement age is May 2000.

However Bill doesn't receive benefits until he has been that age for a full month. So, my understanding is that if Bill files for benefits in April 2000, benefits will be calculated with 2 RF, since he is considered 63 and 10 months old until the last day of April.

If bill files for benefits in May 2000, does he automatically get 0 RF? That seems to be what this  comment suggests:

> NOTE: The throughout the month rule does not apply to the attainment of FRA. Unreduced benefits are payable beginning with the month FRA is attained regardless of the day of the month. For example, if the birth date is June 1, 1935, FRA is attained on May 31, 2000 and unreduced benefits can begin with May 2000.

That's surprising, because the pattern is 4,3,2,0. 

What about June 2000? Is he now receiving 1 delayed credit? (4,3,2,0,1,2) Or 0 delayed credits (4,3,2,0,0,1)?

**Comment by Gregable** [comment]
OK, I think I understand this now. Birthdays on the 1st are for all calculations the previous month. Birthdays on the 2nd are only different from the 2rd in that they can start their first month of benefit on the same month as the birthday. This does not effect RF or DC adjustments. 

RF / DC is always calculated from the MOE / BD difference. Having a birthday in the middle of the month doesn't effect this.

**Comment by stkeros** [comment]
> OK, I think I understand this now. Birthdays on the 1st are for all calculations the previous month. Birthdays on the 2nd are only different from the 2rd **_[I assume you meant THIRD?]_** in that they can start their first month of benefit on the same month as the birthday. This does not effect RF or DC adjustments. **_[I think 2nd vs 3rd DOES affect the reduction factor? See below.]_**
> 
> RF / DC is always calculated from the MOE / BD difference. Having a birthday in the middle of the month doesn't effect this.

I believe you are mostly correct, but am not entirely sure with how you are describing it.

Let's stick with the POMS example of age 62 claiming with an age 65 FRA.

An end of the month birthday behaves identically in all situations as if they were born one day later (e.g. June 30 vs July 1, March 31 vs April 1, etc). No differences for any calculations, only that "SS " birthmonth and "layperson" birthmonths are different. So those two situations can be lumped together.

But for 2nd vs 3rd birthdays. Both can begin to receive benefits in the month of their birth, age 62. But a THIRD birthday gets one extra month reduction. Suppose an age 65 FRA, a 3rd of the month birthday gets the "full" 3 years aka 36 months reduction. Whereas the 1st and 2nd only get 35 months reduction. the 35 month reduction for a 1st of month is because they "turn FRA" in the "previous" month (i.e. June 1st birthday is FRA in May) but they did not take an initial benefit until June (the month they were 62 for the full month). Of course, the 3rd of month birthday, turns FRA in that month (e.g. in June, for June 3rd birthday) but also took a benefit  in June at age 62, a fill 3 years early.

I'm not sure if any of that helped you or confused you more.

**Comment by Gregable** [comment]
> 2nd are only different from the 2rd [I assume you meant THIRD?]

Yes, 3rd.

> But for 2nd vs 3rd birthdays. Both can begin to receive benefits in the month of their birth, age 62.

Doesn't the POMS summary table 4.1 disagree with this statement? Cory and Demi are born (lay birthday) on 6/2/35 and 6/3/35. Cory's "1st MOE" is 6/97 and Demi's is 7/97. Or is your "month of their birth" here referring to the "ss birthday" rather than the "lay birthday"?

My interpretation of that table is that the RF is only different because of the Demi starts benefits a month later than Cory. If Cory and Demi both started benefits in July 97 (due to Cory delaying 1 month more than the minimum), they would both have the same RF.

It sounds like I have that incorrect.

**Comment by stkeros** [comment]
So sorry, I kept using second/third incorrectly based on "order" by mistake. E.g. "third" was the third 'situation' (31st, 1st, second, etc). It's a mess. I have EDITED the comment to correct my errors and to use SS's names (anne/bill, etc) for clarity. And reviewing your comments, it turns out I was responding to something which I thought to be incorrect, but I misread your comment and missed a negative.

Which brings me to, yes, I think your understanding is indeed correct! Sorry for the confusion!

**Comment by Gregable** [comment]
No worries. I think I understand now. Will work on finishing this up soon.

**Comment by Gregable** [comment]
For the "Special Rule" boxes, my plan is to show two different possible boxes, or no box at all.

1. If you were born on the 2nd of the month. I show no "Special Rule" box. You can begin benefits the same month as your 62nd lay birthday and your SSA birthday is the same month as your lay birthday.
2. If you were born on the 1st of the month.  I show a "Special Rule" explaining the "English common law" part about attaining your birthday a day early, explaining why all of the dates in the report are one month early. This box will be shown near where you pick your birthday.
3. If you were born on the 3rd of the month or later. I show a "Special Rule" explaining why you can't start benefits until the month after your birthday, based on the "entire month" rule. This box will be shown in the "Early and Delayed Retirement" and "Early and Delayed Spouse Retirement" sections.

**Comment by Gregable** [comment]
Also, I'm going to set all of the demo birthdates to the 2nd. No reason to complicate the demos.

**Comment by Gregable** [comment]
OK, I think I have it working. For now I set the 3 demos to have 1st, 2nd and 3rd birthdays for ease of testing. If you want, you can also set the birthdays to whatever you want by pasting in the data directly. Just copy the data from these files:

https://socialsecurity/tools/averagepaste.txt
https://socialsecurity/tools/millionpaste.txt
https://socialsecurity/tools/youngpaste.txt

**Comment by stkeros** [comment]
Links take me to:

"This site can’t be reached
socialsecurity’s server IP address could not be found.
Try running Windows Network Diagnostics.
DNS_PROBE_FINISHED_NXDOMAIN"

**Comment by Gregable** [comment]
Oops, typo. I fixed the links.

**Comment by stkeros** [comment]
If spouses b-day set as 2nd, special rule shows up for spouse, but not for self?

Also, the sliders allow one to select an age 62+0 date for spouse with b-day on 2nd, even though the rule says no.

**Comment by Gregable** [comment]
Yes, for example, if you look at the first example link, both Self and Spouse are born on the 1st, and Spouse is born March 1st. So, the slider stops at "62 + 1", which is shown as March. That's the SS age.

The only mention is in the "Normal Retirement Age" section above. 

Here's the logic for the "Special Rule"s:
There are two text boxes:

Box A Text: 

> The month you reach specific ages may seem off by one from what you entered. Social Security follows English common law that finds that a person attains an age on the day before the birthday. Since you were born on the first of the month, this means for example that you attain age 68 on Jun 30, 2018.


Box B Text: 

> You may find it oddly specific that the earliest you can actually begin benefits is at age 62 and 1 month. Benefit eligibility is calculated based on the first month that you are a particular age throughout the entire month. For most, this is the month after their birthday.
>
> For example, the first month that you are 69 throughout the entire month is Apr 2018.

For self: if birthday is on the 1st or later, I show Box A in the "Normal Retirement Age" section.
For self: if birthday is on the 3rd or later, I show Box B in the "Early and Delayed Retirement" section.

For spouse: if birthday is on the 1st or later, I show Box A in the "Early and Delayed Spouse Retirement" section.
For spouse: if birthday is on the 3rd or later, I show Box B immediately above the "Survivor Benefits" section, below the combined benefit chart.

In both of the spouse cases, I suppress the box if the same box was already shown above for Self (ie: if Box A was shown for self, I'll still show Box B for spouse, but not Box A again. I'm not sure if repeating the information twice would be useful or clutter. I suppose it depends on if the user understood if from the first pass or not.

Open to thoughts on the logic or positioning, it's very easy to change.
